### PR TITLE
pyproject based build

### PR DESCRIPTION
**Pull request by clemens-fricke**
Update the build system to use the new standard `pyproject.toml` file. Further support for the python version `3.6` was removed and added for version `3.12`. 

It uses `setuptools` as the build-backend.

The package should now only be installed via `pip install ...`, even for local and editable installs. See `CONTRIBUTING.md` to see the syntax for this operation.

### CODE REVIEW

**Comment by clemens-fricke** [comment]
What I forgot to mention, is that I clipped the `vedo` version so that it must be smaller than `2023.5.0` (version from yesterday) since they renamed `UGrid` to `UnstructuredGrid` (I also could not find this in their release logs, but the change is there). 

At some point, we might need to update all our references to this new version, but currently, I think it is better to clip the version.

**Comment by j042** [comment]
> What I forgot to mention, is that I clipped the `vedo` version so that it must be smaller than `2023.5.0` (version from yesterday) since they renamed `UGrid` to `UnstructuredGrid` (I also could not find this in their release logs, but the change is there).
> 
> At some point, we might need to update all our references to this new version, but currently, I think it is better to clip the version.

Yeah, I guess there's some updates to do ;)

File: pyproject.toml
Comment (j042) [inline pyproject.toml:27]: should we test in 3.12?

File: pyproject.toml
Comment (j042) [inline pyproject.toml:47]: we don't use pytest-cov in CIs

File: pyproject.toml
Comment (j042) [inline pyproject.toml:10]:
```suggestion
```

File: pyproject.toml
Comment (j042) [inline pyproject.toml:4]:
```suggestion
    {name = "Jaewook Lee", email = "jaewooklee042@gmail.com"}
```

**Review by j042 — COMMENTED** [review]
Thank you!

**Comment by clemens-fricke** [comment]
The build and test workflow is still running 3.6 I have to change it before merge.

File: pyproject.toml
Comment (clemens-fricke) [reply pyproject.toml:4]: Sorry

**Comment by clemens-fricke** [comment]
we might want to hold off testing for `3.12` since it needs to build splinepy from source since there is no pypi version yet?

**Comment by j042** [comment]
> we might want to hold off testing for `3.12` since it needs to build splinepy from source since there is no pypi version yet?

we don't need splinepy for gustaf

**Comment by j042** [comment]
good catch in last commit! thanks :)

**Review by j042 — APPROVED** [review]
great, thanks!

**Pull request merged**
